src/d3a/models/strategy/load_hours.py

```diff
@@ -173,6 +142,18 @@ def area_reconfigure_event(self, avg_power_W=None, hrs_per_day=None,
             self.avg_power_W = avg_power_W
             self.assign_energy_requirement(avg_power_W)
 
+        self._area_reconfigure_prices(final_buying_rate)
+
+    def event_activate_price(self):
+        # If use_market_maker_rate is true, overwrite final_buying_rate to market maker rate
+        if self.use_market_maker_rate:
+            self.area_reconfigure_event(final_buying_rate=GlobalConfig.market_maker_rate)
+        self._validate_rates()
+        self.bid_update.update_on_activate()
```

Comment:
Correct me if Im wrong, what I have understood so far is that this bid_price_curve would also be formed for  [LoadExternalStrategy](https://github.com/gridsingularity/d3a/pull/626/files#diff-df43fe29b55cd7a82421a0c25a3f60dfR25) which I think should be avoided as `External Agent` would be having their own algorithm to buy energy

Reply:
The whole method `event_activate_price` is overridden in the `LoadHoursExternalStrategy` , and the base class method is invoked only if the `connected` parameter is true, essentially only when a device is connected to the strategy. The reason it is implemented this way (I would prefer to skip it altogether) is because there is a requirement for the device to follow the normal LoadHoursStrategy until the external client is registered/connected to the device.